universe/supplyverifier/util_test.go

```diff
@@ -0,0 +1,798 @@
+	blockHeight uint32) supplycommit.RootCommitment {
+
+	t.Helper()
+
+	// Create a dummy transaction.
+	tx := &wire.MsgTx{
+		Version: 2,
+		TxIn: []*wire.TxIn{{
+			PreviousOutPoint: wire.OutPoint{
+				Hash:  chainhash.Hash{},
+				Index: 0,
+			},
+		}},
+		TxOut: []*wire.TxOut{{
+			Value:    1000,
+			PkScript: []byte{0x51},
+		}},
+	}
+
+	// Create a dummy internal key.
+	privKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		t.Fatalf("failed to create private key: %v", err)
+	}
+	internalKey := keychain.KeyDescriptor{
+		PubKey: privKey.PubKey(),
+		KeyLocator: keychain.KeyLocator{
+			Family: keychain.KeyFamily(1),
+			Index:  0,
+		},
+	}
+
+	// Create a dummy commitment block.
+	commitmentBlock := supplycommit.CommitmentBlock{
+		Height:    blockHeight,
+		Hash:      chainhash.Hash{},
+		TxIndex:   0,
+		ChainFees: 1000,
+	}
+
+	return supplycommit.RootCommitment{
+		Txn:             tx,
+		TxOutIdx:        0,
+		InternalKey:     internalKey,
+		OutputKey:       privKey.PubKey(),
+		CommitmentBlock: fn.Some(commitmentBlock),
+		SpentCommitment: fn.None[wire.OutPoint](),
+	}
+}
+
+// createTestAssetSpec creates a test asset specifier for testing.
+func createTestAssetSpec(t *testing.T) asset.Specifier {
+	t.Helper()
+
+	privKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		t.Fatalf("failed to create private key: %v", err)
+	}
+
+	return asset.NewSpecifierFromGroupKey(*privKey.PubKey())
+}
+
+// createTestDelegationKey creates a test delegation key for testing.
+func createTestDelegationKey(t *testing.T) btcec.PublicKey {
+	t.Helper()
+
+	privKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		t.Fatalf("failed to create private key: %v", err)
+	}
+	return *privKey.PubKey()
+}
+
+// createTestMetaReveal creates a test meta reveal with delegation key for
+// testing.
+func createTestMetaReveal(t *testing.T) *proof.MetaReveal {
+	t.Helper()
+
+	delegationKey := createTestDelegationKey(t)
+	return &proof.MetaReveal{
+		Data:          []byte("test-metadata"),
+		Type:          proof.MetaOpaque,
+		DelegationKey: fn.Some(delegationKey),
+	}
+}
+
+// createTestMetaRevealWithKey creates a test MetaReveal with a specific
+// delegation key.
+func createTestMetaRevealWithKey(t *testing.T,
+	delegationKey *btcec.PublicKey) *proof.MetaReveal {
+
+	t.Helper()
+
+	return &proof.MetaReveal{
+		Data:          []byte("test-metadata"),
+		Type:          proof.MetaOpaque,
+		DelegationKey: fn.Some(*delegationKey),
+	}
+}
+
+// createTestMintEvent creates a test mint event for testing.
+// Note: This creates a minimal mint event that will fail processing due to
+// missing RawProof, which is useful for testing error conditions.
+func createTestMintEvent(t *testing.T,
+	blockHeight uint32) supplycommit.NewMintEvent {
+
+	t.Helper()
+
+	// Create a dummy script key.
+	privKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		t.Fatalf("failed to create private key: %v", err)
+	}
+	scriptKey := privKey.PubKey()
+
+	// Create a dummy outpoint.
```

Comment:
Superfluous comments here and elsewhere in this file (doesn't add any context re the why/how).